package/MDAnalysis/topology/base.py

```diff
@@ -142,6 +142,42 @@ def squash_by(child_parent_ids, *attributes):
     return atom_idx, unique_resids, [attr[sort_mask] for attr in attributes]
 
 
+def squash_by_attributes(squash_attributes, *other_attributes):
+    """Squash a child-parent relationship using combinations of attributes
+    parents will retain their order of appearance
+
+    Arguments
+    ---------
+    squash_attributes - list of attribute arrays (attributes used to
+                        identify the parent)
+    *other_attributes - other arrays that need to follow the sorting of ids
+
+    Returns
+    -------
+    child_parents_idx - an array of len(child) which points to the index of
+                        parent
+    parent_combos - len(parent) of the unique combinations
+    squashed_attrs - len(parent) of the attributes used for squashing
+    *other_attrs - len(parent) of the other attributes
```

Comment:
Can this follow the doc style a bit better?  Usually it's type annotation on the same line as the variable name and description on the following line

Reply:
This is a good point, I have actually run into an example of this using mol2 generated from pdb in my work because it lost chain information. I’ll add a test for wrapped resids.